subsys/bluetooth/host/l2cap.c

```diff
@@ -795,20 +790,24 @@ static void l2cap_chan_rx_init(struct bt_l2cap_le_chan *chan)
 
 	/* Use existing MTU if defined */
 	if (!chan->rx.mtu) {
-		chan->rx.mtu = L2CAP_MAX_LE_MTU;
+		/* If application has not provide the incoming L2CAP SDU MTU use
+		 * an MTU that does not require segmentation.
+		 */
+		chan->rx.mtu = BT_L2CAP_SDU_RX_MTU;
 	}
 
-	/* MPS shall not be bigger than MTU + 2 as the remaining bytes cannot
-	 * be used.
+	/* MPS shall not be bigger than MTU + BT_L2CAP_SDU_HDR_SIZE as the
+	 * remaining bytes cannot be used.
 	 */
-	chan->rx.mps = MIN(chan->rx.mtu + 2, L2CAP_MAX_LE_MPS);
+	chan->rx.mps = MIN(chan->rx.mtu + BT_L2CAP_SDU_HDR_SIZE,
+			   BT_L2CAP_RX_MTU);
 
 	/* Truncate MTU if channel have disabled segmentation but still have
 	 * set an MTU which requires it.
 	 */
-	if (!chan->chan.ops->alloc_buf && (chan->rx.mps < chan->rx.mtu + 2)) {
-		BT_WARN("Segmentation disabled but MTU > MPS, truncating MTU");
```

Comment:
Not sure why the above warning was removed, just because the documentation states that it would be truncating doesn't mean the developer is aware of that so I think it is still a good practice to warn that something is off.

Reply:
Let's leave it in for now.